cmake/KokkosConfigCommon.cmake.in

```diff
@@ -306,3 +306,38 @@ FUNCTION(kokkos_compilation)
         ENDFOREACH()
     ENDIF()
 ENDFUNCTION()
+
+# this function is provided to easily set which files use the same programming language as Kokkos:
+#
+#       SOURCE      --> specific source files
+#
+FUNCTION(kokkos_set_source_language LANG_SOURCE)
+
+      IF(NOT DEFINED Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
+        MESSAGE(FATAL_ERROR "Using language features requires Kokkos to be build with Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE")
+      ENDIF()
+
+      IF(LANG_SOURCE)
+        MESSAGE(DEBUG "KOKKOS: Setting language to ${Kokkos_COMPILE_LANGUAGE} for SOURCE ${LANG_SOURCE}")
+        SET_PROPERTY(SOURCE ${LANG_SOURCE} PROPERTY LANGUAGE ${Kokkos_COMPILE_LANGUAGE})
+      ENDIF()
+
+ENDFUNCTION()
+
+# this function is provided to easily set the language specific properties of a target to the ones exported by kokkos:
+#
+#       TARGET      --> specific TARGET
+#
+FUNCTION(kokkos_set_target_properties PROP_TARGET)
```

Comment:
The name doesn't reflect that this specifically related to `COMPILE_AS_CMAKE_LANGUAGE`. I would expect that I can use the function either way which means removing the check below.

Reply:
let me check how to make make it independent of `COMPILE_AS_CMAKE_LANGUAGE`